x-pack/plugins/security_solution/scripts/endpoint/endpoint_agent_runner/settings.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import path from 'path';
+import { copyFile } from 'fs/promises';
+import { existsSync } from 'fs';
+import execa from 'execa';
+import { SettingsStorage } from '../common/settings_storage';
+
+export class EndpointAgentRunnerSettings extends SettingsStorage {
+  constructor() {
+    super('settings.json', {
+      directory: path.join(SettingsStorage.getDefaultDirectoryPath(), 'endpoint_agent_runner'),
+    });
+  }
+
+  async setup(): Promise<void> {
+    await this.ensureExists();
+
+    const vmConfigDir = path.resolve(__dirname, 'vm_config');
+
+    // Copy certain files to the settings folder
+    const files: Array<{ from: string; to: string; permissions?: number }> = [
```

Comment:
The SSH key needs to have `600` permission on it in order for `ssh` to use it. So we copy this over to the local user's home directory (`~/.kibanaSecuritySolutionCliTools/endpoint_agent_runner/`) and then change the permissions on the file there.

)FYI: this directory will also be used at some point in the future to keep a directory that is mounted on to the VMs to support faster install/download of the Elastic agent._